### PR TITLE
Use container type to guide search among container status

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In manager#SetContainerReadiness, we call findContainerStatus twice where containers / init containers are iterated.

This PR introduces an enum representing whether the container Id is for container or init container so that in the second call we search the correct type of containers.

```release-note
NONE
```

### CODE REVIEW

**Review by yastij — APPROVED** [review]
/lgtm

Even though the effect of this might not be visible cc @kubernetes/sig-node-pr-reviews

**Comment by tedyu** [comment]
Thanks @yastij 
Do you know who can give approval ?

**Review by mattjmcnaughton — COMMENTED** [review]
I don't feel super strongly about this, but my gut says that the readability hit here isn't worth the slight performance improvements.

The cost of iterating through `status.ContainerStatuses` is bound by the number of containers in the pod, and I wouldn't expect that to ever be meaningfully large enough to make a difference.

I also find the introduction of an `unknown` const slightly confusing, particularly how passing `unknown` as a container type leads to us searching through all the containers in `findContainerStatus`.

Just my opinion :) Definitely curious what others think!

**Comment by tedyu** [comment]
I am open to using another enum in place of unknown.
unknown means we don't know the type of container the container Id represents.

File: pkg/kubelet/status/status_manager.go [hunk 1]
```diff
@@ -190,6 +190,12 @@ func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
 	m.updateStatusInternal(pod, status, pod.DeletionTimestamp != nil)
 }
 
+const (
+	unknown       = iota
+	container     = iota
+	initContainer = iota
```
Comment (krmayankk) [inline pkg/kubelet/status/status_manager.go]: nit: why repeat the `iota` ?

Comment (tedyu) [reply pkg/kubelet/status/status_manager.go]: This is to assign unique values to the 3 identifiers.

**Comment by tedyu** [comment]
@krmayankk 
Can you take another look ?

**Comment by tedyu** [comment]
ping @krmayankk

**Pull request closed without merging**